src/hooks/character.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from 'react';
+import { getApi } from '../services/api.js';
+import { getApiByName } from '../services/api';
+
+export const useCharacters = () => {
+  const [loading, setLoading] = useState(true);
+  const [characters, setCharacters] = useState([]);
+
+  useEffect(() => {
+    getApi()
```

Comment:
you need to pass the page here to getApi, that's why it's only grabbing the first page right now because it defaults to 1 in your services.